lib/middleman-images/image.rb

```diff
@@ -48,21 +88,15 @@ def processed_source_file
 
       private
 
-      def resize(image_path, options)
-        begin
-          require "mini_magick"
-        rescue LoadError
-          raise 'The gem "mini_magick" is required for image resizing. Please install "mini_magick" or remove the resize option.'
-        end
-
-        image_ext = File.extname(image_path)
-        if IGNORE_RESIZING.keys.include? image_ext
-          app.logger.warn("== Images: " + (IGNORE_RESIZING[image_ext] % { file: image_path }))
-          return
-        end
+      def mini_magick(image_path, options)
+        h = {}
+        mini_magick_options = options.slice(*MINI_MAGICK_OPTIONS)
+        ensure_mini_magick(mini_magick_options.keys)
 
         image = MiniMagick::Image.new(image_path) do |i|
-          i.resize(options)
+          mini_magick_options.each_pair do |action, args|
+            i.public_send(action, args) if action_possible?(image_path, action)
```

Comment:
What do you think of the name `ignore_action?` instead of `action_possible?`. I think, it might be more explicit about the intention of the method.
```suggestion
            i.public_send(action, args) unless ignore_action?(image_path, action)
```

Reply:
Personally I don't like method names with negations in their name. And somehow ignoring something feels a bit negated to me. But I renamed it.